### PR TITLE
feat(git-tokens): add create/edit flow

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-871

- Add Git Tokens for providers
- Add 4 new hooks and using React Query: `useAuthProviders`, `useGitTokens`, `useRepositories`, and `useBranches` 
- Updated flow create and edit service with Git provider

Capture d’écran 2023-10-25 à 17 38 42

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(git-tokens): add create flow - 2023-10-23T14:58:27Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/250d4e4c-ef8a-4dab-9056-061c590848cf/services/general)

 Another comment will be posted when deployments are terminated

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://913-console-staging.zc531a994.rustrocks.cloud)

File: libs/domains/organizations/feature/src/lib/hooks/use-git-tokens/use-git-tokens.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useQuery } from '@tanstack/react-query'
+import { queries } from '@qovery/state/util-queries'
+
+export interface UseGitTokensProps {
+  organizationId: string
+}
+
+export function useGitTokens({ organizationId }: UseGitTokensProps) {
+  return useQuery({
+    // eslint-disable-next-line @typescript-eslint/no-extra-non-null-assertion
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-git-tokens/use-git-tokens.tsx:10]: you can remove this

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 2]
```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
       return response.data
     },
   }),
+  gitTokens: (organizationdId) => ({
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:28]: types are missing and we should always use object to pass params
```suggestion
  gitTokens: ({ organizationdId } : { organizationdId: string }) => ({
```
Valid for all methods here

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 3]
```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
...
+    queryKey: [organizationId],
+    async queryFn() {
+      const response = await gitApi.getOrganizationGitProviderAccount(organizationId)
+      return response.data
+    },
+  }),
+  repositories: (organizationId, gitProvider, gitToken) => ({
+    queryKey: [organizationId, gitProvider, gitToken],
+    async queryFn() {
+      if (gitProvider === GitProviderEnum.GITHUB) {
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:45]: If you type `gitProvider` as `gitProvider: keyof typeof GitProviderEnum` you can use ts-pattern `match` and be exhaustive.

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 4]
```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
...
+      if (gitProvider === GitProviderEnum.GITLAB) {
+        const response = await gitApi.getOrganizationGitlabRepositories(organizationId, gitToken)
+        return response.data
+      }
+      if (gitProvider === GitProviderEnum.BITBUCKET) {
+        const response = await gitApi.getOrganizationBitbucketRepositories(organizationId, gitToken)
+        return response.data
+      }
+
+      return Promise.all([])
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:58]: This is not correct because it's not the same return type. Using strong types as params, this should not be necessary

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 5]
```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
...
+    },
+  }),
+  branches: (organizationId, gitProvider, name, gitToken) => ({
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:61]: same comments as `repositories`

File: libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx:1]: rename git-branch-settings

File: libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    enabled: !disabled,
+  })
+
+  if (isError) {
+    return null
+  }
+
+  if (!disabled && (isLoading || isRefetching)) {
+    return (
+      <div data-testid="loader" className="flex justify-center">
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx:41]: Is this data-testid still needed?

File: libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        <LoaderSpinner />
+      </div>
+    )
+  } else {
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx:45]: no need for `else` with `if` with early returns

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 8]
```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
...
+    async queryFn() {
+      const response = await organizationApi.listOrganizationGitTokens(organizationdId)
+      return response.data.results
+    },
+  }),
+  authProviders: (organizationId) => ({
+    queryKey: [organizationId],
+    async queryFn() {
+      const response = await gitApi.getOrganizationGitProviderAccount(organizationId)
+      return response.data
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:39]:
```suggestion
      return response.data.results
```

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts [hunk 9]
```diff
@@ -18,6 +25,58 @@ export const organizations = createQueryKeys('organizations', {
...
+    },
+  }),
+  repositories: (organizationId, gitProvider, gitToken) => ({
+    queryKey: [organizationId, gitProvider, gitToken],
+    async queryFn() {
+      if (gitProvider === GitProviderEnum.GITHUB) {
+        const response = await gitApi.getOrganizationGithubRepositories(organizationId, gitToken)
+        return response.data
```
Comment (ctjhoa) [inline libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:47]:
```suggestion
        return response.data.results
```
For all cases

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+  const providerOptions = disabled
+    ? [
+        {
+          label: watchFieldProvider ?? '',
+          value: watchFieldProvider ?? '',
+          icon: <Icon name={`${watchFieldProvider?.split(' ')[0].toUpperCase()}`} />,
+        },
+      ]
+    : mergeProviders(authProviders as GitAuthProvider[], gitTokens)
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx:52]:
```suggestion
    : mergeProviders(authProviders, gitTokens)
```
Cast should not be needed

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    value: provider.name || '',
+    icon: <Icon width={16} height={16} name={provider.name} />,
+  }))
+
+  const currentGitTokens =
+    gitTokens?.map((token: GitTokenResponse) => ({
+      label: upperCaseFirstLetter(token.name),
+      value: `TOKEN_${token.type}_${token.id}`,
+      icon: <Icon width={16} height={16} name={token.type} />,
+    })) || []
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx:25]: style:
if you assign `gitTokens = []` in the params you can simplify code

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+import { upperCaseFirstLetter } from '@qovery/shared/util-js'
+import useAuthProviders from '../hooks/use-auth-providers/use-auth-providers'
+import useGitTokens from '../hooks/use-git-tokens/use-git-tokens'
+
+export interface GitProviderSettingProps {
+  disabled?: boolean
+}
+
+export const mergeProviders = (authProviders: GitAuthProvider[], gitTokens?: GitTokenResponse[]) => {
+  const currentAuthProviders = authProviders.map((provider: GitAuthProvider) => ({
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx:14]:
```suggestion
  const currentAuthProviders = authProviders.map((provider) => ({
```
Type should not be needed here

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+    label: `${upperCaseFirstLetter(provider.name)} (${provider.owner})`,
+    value: provider.name || '',
+    icon: <Icon width={16} height={16} name={provider.name} />,
+  }))
+
+  const currentGitTokens =
+    gitTokens?.map((token: GitTokenResponse) => ({
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx:21]:
```suggestion
    gitTokens?.map((token) => ({
```
Type should not be needed here

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+    enabled: !disabled,
+  })
+
+  if (isError) {
+    return null
+  }
+
+  if (!disabled && (isLoading || isRefetching)) {
+    return (
+      <div data-testid="loader" className="flex justify-center">
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:40]: Is this data-testid still needed?

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+export interface GitRepositorySettingProps {
+  gitProvider: GitProviderEnum | string
+  disabled?: boolean
+}
+
+export function GitRepositorySetting({ disabled, gitProvider }: GitRepositorySettingProps) {
+  const { control, setValue, watch } = useFormContext()
+  const { organizationId = '' } = useParams()
+
+  const getGitToken = getGitTokenValue(gitProvider)
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:18]:
```suggestion
  const gitToken = getGitTokenValue(gitProvider)
```

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+    return null
+  }
+
+  if (!disabled && (isLoading || isRefetching)) {
+    return (
+      <div data-testid="loader" className="flex justify-center">
+        <LoaderSpinner />
+      </div>
+    )
+  } else {
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:44]: no need for `else` with `if` with early returns

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+            options={
+              disabled
+                ? [
+                    {
+                      label: upperCaseFirstLetter(watchFieldRepository) ?? '',
+                      value: watchFieldRepository ?? '',
+                    },
+                  ]
+                : // TODO: need to fix api doc to remove "as"
+                  (repositories as GitRepository[])?.map((repository) => ({
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:64]: I don't think it's API doc issue here but it's rather related to return `data` instead of `data.results` (cf previous comment)

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+                    label: upperCaseFirstLetter(repository.name),
+                    value: repository.name,
+                  }))
+            }
+            onChange={(event) => {
+              field.onChange(event)
+              const currentRepository = (repositories as GitRepository[]).find(
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:71]: no cast needed here

File: libs/shared/console-shared/src/lib/git-repository-settings/ui/confirmation-git-modal/confirmation-git-modal.tsx [hunk 19]
```diff
@@ -44,16 +42,16 @@ export function ConfirmationGitModal(props: ConfirmationGitModalProps) {
...
         >
           Cancel
         </ButtonLegacy>
         <ButtonLegacy
           dataTestId="submit-button"
           size={ButtonLegacySize.XLARGE}
           onClick={() => {
-            props.onSubmit && props.onSubmit()
-            props.onClose()
+            onSubmit && onSubmit()
```
Comment (ctjhoa) [inline libs/shared/console-shared/src/lib/git-repository-settings/ui/confirmation-git-modal/confirmation-git-modal.tsx:53]:
```suggestion
            onSubmit()
```

File: libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.spec.tsx [hunk 20]
```diff
@@ -115,6 +115,9 @@ describe('PageSettingsGeneral', () => {
     const button = screen.getByTestId('submit-button')
 
     await userEvent.click(button)
-    expect(spy).toHaveBeenCalled()
+
+    waitFor(() => {
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.spec.tsx:119]: Why is this needed now?

File: libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export function useRepositories({ organizationId, gitProvider, enabled, gitToken }: UseRepositoriesProps) {
+  return useQuery({
+    ...queries.organizations.repositories(organizationId, gitProvider, gitToken),
+    meta: {
+      notifyOnError: true,
+    },
+    enabled: enabled,
+    refetchOnWindowFocus: false,
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx:19]: This should not be added to all `useQuery`. This was needed only for WS because it's not a simple HTTP query

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx (hunk 21 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx:19]: I added it for these queries (auth provider, repository, branches), because when we come back on the browser we have an unnecessary reload with the spinner displayed. I think is not necessary because the data has not changed a lot from the API

File: libs/shared/console-shared/src/lib/git-repository-settings/ui/confirmation-git-modal/confirmation-git-modal.tsx (hunk 19 above)
Comment (RemiBonnet) [reply libs/shared/console-shared/src/lib/git-repository-settings/ui/confirmation-git-modal/confirmation-git-modal.tsx]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx (hunk 17 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx]: The data is returned on the `response.data` that's why I need to fix the API doc

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx (hunk 15 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:18]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx (hunk 14 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx]: Not needed, removed https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx (hunk 13 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx (hunk 12 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx (hunk 11 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx (hunk 10 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx:51]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx (hunk 7 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx]: https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx (hunk 6 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx]: Not needed, removed https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts (hunk 5 above)
Comment (RemiBonnet) [reply libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts]: Fixed https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts (hunk 2 above)
Comment (RemiBonnet) [reply libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts]: Fixed https://github.com/Qovery/console/pull/913/commits/5f0ff27fbf70a12706bd3d915ea9360cbec3cc55

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts (hunk 4 above)
Comment (RemiBonnet) [reply libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts]: https://github.com/Qovery/console/pull/913/commits/5b763f82917abfa2a67dcdf0b1a9db7ccb4c77d6

File: libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-branch-setting/git-branch-setting.tsx:1]: https://github.com/Qovery/console/pull/913/commits/211909a284d7cfc2f877d891ed02b02eed2f531b

File: libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts (hunk 8 above)
Comment (RemiBonnet) [reply libs/domains/organizations/data-access/src/lib/domains-organizations-data-access.ts:41]: The API response returns the data on the `response.data`, not on the `response.data.results` I'll see with back-end if it's normal

File: libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.spec.tsx (hunk 20 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/ui/page-settings-general/page-settings-general.spec.tsx:119]: I have the feeling because with set values for Repository and Branches after setting the Auth Providers, the DOM needed to wait for the Auth Providers values before validating others, that's why I added this

File: libs/domains/organizations/feature/src/lib/hooks/use-auth-providers/use-auth-providers.tsx [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface UseAuthProvidersProps {
+  organizationId: string
+  enabled?: boolean
+}
+
+export function useAuthProviders({ organizationId, enabled }: UseAuthProvidersProps) {
+  return useQuery({
+    ...queries.organizations.authProviders({ organizationId }),
+    enabled: enabled,
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-auth-providers/use-auth-providers.tsx:12]:
```suggestion
    enabled,
```

File: libs/domains/organizations/feature/src/lib/hooks/use-branches/use-branches.tsx [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+  gitProvider: GitProviderEnum
+  name: string
+  gitToken?: string
+  enabled?: boolean
+}
+
+export function useBranches({ organizationId, gitProvider, name, gitToken, enabled }: UseBranchesProps) {
+  return useQuery({
+    ...queries.organizations.branches({ organizationId, gitProvider, name, gitToken }),
+    enabled: enabled,
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-branches/use-branches.tsx:16]:
```suggestion
    enabled,
```

File: libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+  enabled?: boolean
+}
+
+export function useRepositories({ organizationId, gitProvider, enabled, gitToken }: UseRepositoriesProps) {
+  return useQuery({
+    ...queries.organizations.repositories({ organizationId, gitProvider, gitToken }),
+    meta: {
+      notifyOnError: true,
+    },
+    enabled: enabled,
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx:18]:
```suggestion
    enabled,
```

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+
+  const providerOptions = disabled
+    ? [
+        {
+          label: watchFieldProvider ?? '',
+          value: watchFieldProvider ?? '',
+          icon: <Icon name={`${watchFieldProvider?.split(' ')[0].toUpperCase()}`} />,
+        },
+      ]
+    : mergeProviders(authProviders as GitAuthProvider[], gitTokens)
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx:51]: If this cast is still needed due to buggy API can you cast `data-access` side like `repositories`. So we can isolate quirky typing at the same place.

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx [hunk 26]
```diff
@@ -0,0 +1,85 @@
...
+export interface GitRepositorySettingProps {
+  gitProvider: GitProviderEnum
+  disabled?: boolean
+}
+
+export function GitRepositorySetting({ disabled, gitProvider }: GitRepositorySettingProps) {
+  const { control, setValue, watch } = useFormContext()
+  const { organizationId = '' } = useParams()
+
+  const getGitToken = getGitTokenValue(gitProvider)
```
Comment (ctjhoa) [inline libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx:18]: You only fix `git-branch-settings`

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/organizations/feature/src/lib/git-branch-settings/git-branch-settings.tsx [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+  const {
+    data: branches = [],
+    isError,
+    isLoading,
+    isRefetching,
+  } = useBranches({
+    organizationId,
+    gitProvider: provider,
+    name: watchFieldRepository,
+    gitToken: gitToken?.id,
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-branch-settings/git-branch-settings.tsx:31]: The request needs a `gitToken` param, I think it's better to keep it

```const response = await gitApi.getOrganizationGithubRepositoryBranches(organizationId, name, gitToken)```

File: libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx (hunk 26 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-repository-setting/git-repository-setting.tsx]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

File: libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx (hunk 25 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-provider-setting/git-provider-setting.tsx]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

File: libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx (hunk 24 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/hooks/use-repositories/use-repositories.tsx]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

File: libs/domains/organizations/feature/src/lib/hooks/use-branches/use-branches.tsx (hunk 23 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/hooks/use-branches/use-branches.tsx]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

File: libs/domains/organizations/feature/src/lib/hooks/use-auth-providers/use-auth-providers.tsx (hunk 22 above)
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/hooks/use-auth-providers/use-auth-providers.tsx]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

File: libs/domains/organizations/feature/src/lib/git-branch-settings/git-branch-settings.tsx (hunk 27 above)
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/lib/git-branch-settings/git-branch-settings.tsx:31]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

File: libs/domains/organizations/feature/project.json [hunk 28]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "domains-organizations-feature",
+  "$schema": "../../../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "libs/domains/organizations/feature/src",
+  "projectType": "library",
+  "tags": [],
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/domains/organizations/feature/project.json]: https://github.com/Qovery/console/pull/913/commits/930209f4f292c911acb87bf6dada6560cea35ae1

**Review by ctjhoa — APPROVED** [review]
